### PR TITLE
Dargwa

### PR DESCRIPTION
**Pull request by d97hah**
I know this PR contains revertions on five iso-codes and some of the scripts but this is the only way the stupid git will let me push to the branch and create the PR. I wasted several hours already, better xrotwang fixes or I do once this branch is over.

### CODE REVIEW

File: languoids/glottocodes.json [hunk 1]
```diff
@@ -9140,6 +9140,7 @@
     "unal": 1234,
     "unam": 1242,
     "unat": 1236,
+    "uncl": 1508,
```
Comment (xrotwang) [inline languoids/glottocodes.json]: Hm. That looks like some weird shit going on here. There must never be duplicate keys in this dictionary - and I guess that happened when you merged some unspecified state of master into this branch.

**Comment by xrotwang** [comment]
Hm, running `git status` and `git diff` often may, and in particular before any bulk `add` and `commit` is highly recommended.

**Comment by xrotwang** [comment]
Note the files here https://github.com/clld/glottolog/pull/270/commits/86c5859d0bb07fca80f50f6d10cc2eb3ccd65770 which have been created when git encountered a merge conflict, `md.ini~HEAD`, et al. This must be cleaned up before commiting.

**Comment by xrotwang** [comment]
So, `glottolog check` passe without errors now, so I'd consider this a consistent state of the tree.

**Comment by d97hah** [comment]
Thanks! I really thought I was doing everything right but then git wizardry
started...

Pada tanggal Sel, 6 Nov 2018 pukul 17.07 Robert Forkel <
notifications@github.com> menulis:

**Comment by xrotwang** [comment]
At least fubar cannot really happen :)

Am Di., 6. Nov. 2018, 18:03 hat d97hah <notifications@github.com>
geschrieben:

**Comment by xflr6** [comment]
Well, looks like the merge did break `treedb.py`, fixed in 16be7513cca81b338eece13a62ef42e4b33f5b4a :)

**Comment by xrotwang** [comment]
Well, looks like `treedb.py` is lacking tests :)

**Comment by xrotwang** [comment]
@d97hah maybe this is an incentive to switch to branch protection for `master`: This would have prevented me and @xflr6 sneaking in commits in `master` while your PR wasn't merged. The alternative would be for you to learn [git rebase](https://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/).

**Comment by xflr6** [comment]
> Well, looks like treedb.py is lacking tests :)

Hehe. I think that was intended so far for `scripts/`. Still wondering how this happened if none of the commits on the other merged branches even toched the directory.

> maybe this is an incentive to switch to branch protection

Hm, requiring to do everything via PRs makes the need for rebasing or merging more visible, but if you have a long-standing branch and some PRs have been merged in the mean time, you still need to rebase/merge, right?

> The alternative would be for you to learn git rebase

For me, using a graphical tool, i.e. https://tortoisegit.org/, makes this much easier.

**Comment by xrotwang** [comment]
@xflr6 the idea would be to not have long-lasting branches, and basically put Harald's PRs into a priority queue :)

**Comment by xrotwang** [comment]
@xflr6 regarding how this happened: My guess is that at some point Harald realized he'd have to merge master, but did so without first pulling master from origin.

**Comment by d97hah** [comment]
Don't see this as a argument for MB protection at all, if anything the
opposite. We are supposed to make changes with PRs, but this is a slow
and cumbersome way to get the updates done so people are tempted to
push directly to master. There are bound to be long branches since
every now and then people don't have the time to (dis)approve PRs
quickly. Merging branches with git is a pain in the ass whenever the
ABC situaton does not apply (though I have not tried the GUI). If
everyone pushed to master no branch merging problems would exist and
I think this is world be better setup than with PRs. Thanks to git we
do have the possibility to go back to earlier states if someone pushes
something bad to master.

Pada tanggal Rab, 7 Nov 2018 pukul 09.18 Robert Forkel <
notifications@github.com> menulis:

**Comment by xrotwang** [comment]
PRs may be slow, but they are not more cumbersome than tripping over an error weeks after it has been introduced and trying to fix it. Pushing code directly to master is sort-of ok, if the code has tests, which will be run upon push. And trading merging issues for silently introduced reverts also doesn't sound like a good plan. So I guess we'll leave the workflow as is :)

**Comment by d97hah** [comment]
What I am saying is PRs *introduce* errors that would not have been there
without PRs because the errors relate to the branching issues and the
volatility of making further changes while previous changes are still
pending (in this case I even got some silently introduced reverts). I don't
see the principled difference between code with tests and data with
glottolog check. Isn't the real reason for the last set of code pushes that
the change(s) were innocuous and that PR would have slowed it down?

Pada tanggal Rab, 7 Nov 2018 pukul 12.51 Robert Forkel <
notifications@github.com> menulis:

**Comment by d97hah** [comment]
I don't think it's a matter of "learning merging with git". You can know
merging with git inside out and still, as a human, get lost in the maze of
rebases, aborts, merges, adds etc. Just like you can know the programming
language Python but not understand how to quickly modify some given Python
code to get what you want. Sure, one can go through everything step-by-step
and then it resolves, but that takes a long time. I am not an example of
someone who knows git merging, but if I were, there's still a risk with PRs
that I make mistakes forgetting what influences what and so on (that I
wouldn't have if there was just one master branch). Just like my Python
code has bugs even though I know this programming language.

You are right in principle about the principled different between
semantics-tested versus syntactic-tested materials, but the semantics-tests
are not complete and the tests themselves may have errors, so it's still
going to be a matter of minor vs major change, or trusted vs untrusted or
something practical like that.



Pada tanggal Rab, 7 Nov 2018 pukul 14.24 Robert Forkel <
notifications@github.com> menulis:

**Pull request merged**